features/graph/documentation/partner-onboarding.md

```diff
@@ -0,0 +1,312 @@
+    "title": "Access your followers and who you are following.",
+    "category": "Follow",
+    "isDefault": true,
+    "isRequired": false,
+  },
+  {
+    "scope": "user-library-modify",
+    "display": "user-library-modify",
+    "description": "Write/delete access to a user's \"Your Music\" library.",
+    "title": "Manage your saved content.",
+    "category": "Library",
+    "isDefault": false,
+    "isRequired": false,
+  },
+  {
+    "scope": "user-read-playback-position",
+    "display": "user-read-playback-position",
+    "description": "Read access to a user’s playback position in a content.",
+    "title": "Read your position in content you have played.",
+    "category": "Listening History",
+    "isDefault": false,
+    "isRequired": false,
+  },
+  {
+    "scope": "playlist-read-private",
+    "display": "playlist-read-private",
+    "description": "Read access to user's private playlists.",
+    "title": "Access your private playlists.",
+    "category": "Playlists",
+    "isDefault": true,
+    "isRequired": false,
+  },
+  {
+    "scope": "user-read-email",
+    "display": "user-read-email",
+    "description": "Read access to user’s email address.",
+    "title": "Get your real email address.",
+    "category": "Users",
+    "isDefault": true,
+    "isRequired": false,
+  },
+  {
+    "scope": "user-read-private",
+    "display": "user-read-private",
+    "description": "Read access to user’s subscription details (type of user account).",
+    "title": "Access your subscription details.",
+    "category": "Users",
+    "isDefault": true,
+    "isRequired": false,
+  },
+  {
+    "scope": "user-library-read",
+    "display": "user-library-read",
+    "description": "Read access to a user's library.",
+    "title": "Access your saved content.",
+    "category": "Library",
+    "isDefault": true,
+    "isRequired": false,
+  },
+  {
+    "scope": "playlist-read-collaborative",
+    "display": "playlist-read-collaborative",
+    "description": "Include collaborative playlists when requesting a user's playlists.",
+    "title": "Access your collaborative playlists.",
+    "category": "Playlists",
+    "isDefault": false,
+    "isRequired": false,
+  },
+  {
+    "scope": "streaming",
+    "display": "streaming",
+    "description": "Control playback of a Spotify track. This scope is currently available to the Web Playback SDK. The user must have a Spotify Premium account.",
+    "title": "Play content and control playback on your other devices.",
+    "category": "Playback",
+    "isDefault": false,
+    "isRequired": false,
+  },
+]
+```
+
+## Graph API/Explorer integration
+
+By integrating with the Graph API/Explorer experience, partners can enable developers to interact with their Graph API surface through Netlify Graph, and combine the information from their API with any of the supported services that are already part of the Netlify Graph.
+
+There are two classes of integrations that we support - basic and advanced.
```

Comment:
I don't like `basic`/`advanced` at all - we *really* need to make the "GID, Node Interface, and Cursor Connection Spec"  the _standard_, really the bare-minimum, and the naming should reflect that.

If we end up with 100 "basic" graphs, the majority of the medium/long-term value of Graph will be lost forever.

Reply:
Open to suggestions here. If indeed we are treating these as "dev" and "production" then we should call them out as such. For example, we will not onboard into the marketplace anything that is not at the production level, but you can still experiment with the dev/test mode. How's that sound?

Reply:
I agree not sure we need "Basic integration:" and "Advanced Integration" at all in the heading. We already explain the justification and trade offs for anyone doing the _advanced_ work: `The above requires more work on the partner side, but as a result... `

Can we provide a similar justification for the "(Optional) UserInfo endpoint: ". "Optional" means easy to not do, should we still call it "Optional"?

Reply:
>we will not onboard into the marketplace anything that is not at the production level

I think we should be more assertive in the docs towards the "advanced integration" but not make it the requirement for getting listed in any marketplace.

My issue is that currently the docs make it seem that basic and advanced are equally encouraged. Would rather default to advanced and keep the note at bottom that explains the tradeoffs for any partner that doesn't have the Node Interface implemented.

Reply:
How would the differences impact devs using the Marketplace? And how would we communicate that impact? This appears to me to be a discussion of quality vs quantity.

Based on my (admittedly limited) understanding, I think treating the "advanced" integration as default makes sense. We can include a note that mentions a more limited form of integration, but discourages it by pointing out disadvantages. It could link to another page that has the limited integration info so that we don't give it much screen real estate here.

For example:

## Integration requirements: GID, Node Interface, and Cursor Connection Spec
- [Global Object Identification](https://graphql.org/learn/global-object-identification/) implemented. We need this for the Node Interface [https://dev.to/zth/the-magic-of-the-node-interface-4le1](https://dev.to/zth/the-magic-of-the-node-interface-4le1). Note that we *could* allow an optional field to be specified for an external API and used in place of `id`, e.g. `nodeId`, but it would be preferable to normalize all of the input APIs outside of the core of Netlify Graph.
- [Cursor Connection Spec](https://dev.to/zth/connection-based-pagination-in-graphql-2588) implemented. It standardizes pagination, and will enable us to generate automatic functions, e.g. `query.users.pageInfo.hasNextPage ? query.users.fetchMore() : ()` and even automate pagination across polling queries to turn non-live APIs on the backend into live-ish APIs that don't drop any items across polling events. It's also what enables automatic nested pagination here in [this example](https://youtu.be/qkkRss6x5ko?t=377), but that requires *both* the GID+Node Interface *and* the Cursor Connection Spec.
The above requires more work on the partner side, but as a result no developer ever writes pagination code (even nested pagination), we can poll stale-APIs efficiently without missing items, and we can build more advanced future capabilities.

---
**NOTE**

Netlify Graph supports [a more limited integration](link-to-different-page) if you can't support the requirements listed above. However, we discourage using this approach because x, y, z reasons.

---

Reply:
Also, I think this info should be near the top of the doc. There's not much point reading the rest of the page if partners don't meet the integration requirements.

Reply:
@KyleBlankRollins @sgrove @safoo I just pushed an update to change how we refer to the "tiers" - let me know what you think.